src/widget/slidereventhandler.h

```diff
@@ -0,0 +1,198 @@
+            }
+
+            // Update display
+            pWidget->update();
+        }
+    }
+
+    void mousePressEvent(T* pWidget, QMouseEvent* e) {
+        if (!m_bEventWhileDrag) {
+            m_dStartMousePos = 0;
+            m_dStartHandlePos = 0;
+            pWidget->mouseMoveEvent(e);
+            m_bDrag = true;
+        } else {
+            if (e->button() == Qt::RightButton) {
+                pWidget->resetControlParameter();
+                m_bRightButtonPressed = true;
+            } else {
+                if (m_bHorizontal) {
+                    m_dStartMousePos = e->x() - m_dHandleLength / 2;
+                } else {
+                    m_dStartMousePos = e->y() - m_dHandleLength / 2;
+                }
+                m_dStartHandlePos = m_dPos;
+            }
+        }
+    }
+
+    void mouseReleaseEvent(T* pWidget, QMouseEvent* e) {
+        if (!m_bEventWhileDrag) {
+            pWidget->mouseMoveEvent(e);
+            m_bDrag = false;
+        }
+        if (e->button() == Qt::RightButton) {
+            m_bRightButtonPressed = false;
+        } else {
+            pWidget->setControlParameter(m_dOldParameter);
+        }
+    }
+
+    void wheelEvent(T* pWidget, QWheelEvent* e) {
+        // For legacy (MIDI) reasons this is tuned to 127.
+        double wheelAdjustment = ((QWheelEvent *)e)->delta() / (120.0 * 127.0);
+        double newParameter = pWidget->getControlParameter() + wheelAdjustment;
+
+        // Clamp to [0.0, 1.0]
+        newParameter = math_clamp_unsafe(newParameter, 0.0, 1.0);
+
+        pWidget->setControlParameter(newParameter);
+        onConnectedControlChanged(pWidget, newParameter);
+        pWidget->update();
+        e->accept();
+    }
+
+    void onConnectedControlChanged(T* pWidget, double dParameter) {
+        // WARNING: The second parameter to this method is unused and called with
+        // invalid values in parts of WSliderComposed. Do not use it unless you fix
+        // this.
+
+        // We don't update slider values while you're dragging them. This way you
+        // don't have to "fight" with a controller that is also changing the
+        // control.
+        if (m_bDrag) {
+            return;
+        }
+
+        if (m_dOldParameter != dParameter) {
+            m_dOldParameter = dParameter;
+
+            double newPos = parameterToPosition(dParameter);
+
+            // Clamp to [0.0, sliderLength - m_dHandleLength].
+            newPos = math_clamp_unsafe(newPos, 0.0, m_dSliderLength - m_dHandleLength);
+
+            // Check a second time for no-ops. It's possible the parameter changed
+            // but the visible pixmap didn't. Only update() the widget if we're
+            // really sure we need to since this involves painting ALL of its
+            // parents.
+            if (newPos != m_dPos) {
+                m_dPos = newPos;
+                pWidget->setControlParameter(dParameter);
```

Comment:
Removed in https://github.com/mixxxdj/mixxx/commit/d2fb971b2b81c3e1f5ef7f7418277b392f0edc7c